File: LeetCode/C++/739.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        vector<int> answer(temperatures.size(), 0);

        stack<int> stk;
        stk.push(0);

        for(int i = 1; i < temperatures.size(); i++) {
            while(!stk.empty() && temperatures[stk.top()] < temperatures[i]) {
                answer[stk.top()] = i - stk.top();
                stk.pop();
            }
            stk.push(i);
        }

        return answer;
    }
};

File: LeetCode/JAVA/739.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        int[] answer = new int[temperatures.length];
        Stack<Integer> stk = new Stack<>();
        stk.push(0);

        for(int i = 1; i < temperatures.length; i++) {
            while(!stk.isEmpty() && temperatures[stk.peek()] < temperatures[i]) {
                answer[stk.peek()] = i - stk.peek();
                stk.pop();
            }
            stk.push(i);
        }

        return answer;
    }
}